action.yml

```diff
@@ -1,27 +1,35 @@
 name: "Setup Scala with both Java and sbt"
 description: "Much longer description"
-#inputs:
-#  sbt-runner-version:
-#    description: "The runner version (The actual version is controlled via project/build.properties)"
-#    required: true
-#    default: 1.10.2
+inputs:
+  major-java-version:
+    description: 'Specify a major Java version (21, 11, etc) as an alternative to reading one from .tool-versions'
+    required: false
+    default: ''
 runs:
   using: "composite"
   steps:
     - name: Determine major Java version
       id: determine-java-version
       shell: bash
+      env:
+        MAJOR_JAVA_VERSION: ${{ inputs.major-java-version }}
       run: |
-        if [ ! -f .tool-versions ]; then
-          echo "::error title=Missing .tool-versions file::setup-scala requires an asdf-format .tool-versions file to establish the Java version for the build."
-          exit 1
+        if [ ${MAJOR_JAVA_VERSION:-} ]; then
+          echo "Reading Java version from the 'major-java-version' input property, rather than .tool-versions"
+        else
+          if [ ! -f .tool-versions ]; then
+            echo "::error title=Missing .tool-versions file::setup-scala requires an asdf-format .tool-versions file to establish the Java version for the build."
```

Comment:
Should this mention the option of the `major-java-version` parameter but suggest that `.tools-version` is preferred?

Reply:
👍  Done with f8b37d4!